### PR TITLE
extract DiscoverySettings to a separate message

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1186

### CODE REVIEW

File: changelog/v0.11.4/discovery-setting.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
```
Comment (joekelley) [inline changelog/v0.11.4/discovery-setting.yaml:2]: Let's mark this as breaking since it restructures the settings API

File: pkg/mesh-discovery/agent/discovery_reconciler.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
+package agent
```
Comment (joekelley) [inline pkg/mesh-discovery/agent/discovery_reconciler.go:1]: This file is unused and appears to be out of scope for this PR

File: changelog/v0.11.4/discovery-setting.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+    issueLink: https://github.com/solo-io/gloo-mesh-enterprise/issues/98
```
Comment (joekelley) [inline changelog/v0.11.4/discovery-setting.yaml:3]: could you create and link to an issue in this repo instead?

File: pkg/mesh-discovery/agent/discovery_reconciler.go (hunk 2 above)
Comment (yuval-k) [reply pkg/mesh-discovery/agent/discovery_reconciler.go:1]: right! copy paste error

File: changelog/v0.11.4/discovery-setting.yaml (hunk 3 above)
Comment (yuval-k) [reply changelog/v0.11.4/discovery-setting.yaml:3]: will do. what's the advantage of doing so?

**Review by ilackarms — DISMISSED** [review]

File: changelog/v0.11.4/discovery-setting.yaml (hunk 3 above)
Comment (joekelley) [reply changelog/v0.11.4/discovery-setting.yaml]: Thanks! This way non-solo users will be able to access a working issue link in the generated release notes, as in `#1167` here https://github.com/solo-io/gloo-mesh/releases/tag/v0.11.2.

Not very impactful in this case, but some issues might have extensive discussion that's worth referencing.

**Review by joekelley — APPROVED** [review]
